### ISSUE TITLE
[web] setSemanticsEnabled(true) makes TextField with `obscureText:true` print out obscured text

### ISSUE THREAD

**Issue by FallenRiteMonk** [flutter/flutter#90794]
With the following line after a runApp call
`RendererBinding.instance!.setSemanticsEnabled(true);`
TextFields with `obscureText: true` are unusable, due to `onChange` and `textController.text` always return obscured text on web.

This behavior was noticed once I updated from 2.2.3 to 2.5.0 and still persists in 2.5.1.

The below example shows the described problem. Removing `RendererBinding.instance!.setSemanticsEnabled(true);` results in TextField returning to desired effect.

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() { 
  runApp(MyApp());
  RendererBinding.instance!.setSemanticsEnabled(true);
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  final String title;

  const MyHomePage({
    Key? key,
    required this.title,
  }) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  String? _text;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Text(
              'You have entered: $_text',
            ),
            TextFormField(
            obscureText: true,
            onChanged: (text) {
              print('Entered: $text');
              setState(() {
                _text = text;
              });
            },
          )
          ],
        ),
      ),
    );
  }
}
```

**Comment by danagbemava-nc** [comment]
Issue is reproducible on `master` and `stable 2.5.1`. It doesn't reproduce on `stable 2.2.3`, so it's likely it is a regression. 

<details>
<summary>screenshots</summary>

`stable 2.2.3`

<img width="697" alt="Screenshot 2021-09-27 at 11 34 52" src="https://user-images.githubusercontent.com/88313112/134918306-62567338-b5c7-4d1a-82d5-587c5334d4d9.png">

`stable 2.5.1 & master`

<img width="801" alt="Screenshot 2021-09-27 at 13 22 43" src="https://user-images.githubusercontent.com/88313112/134918780-ae4cd4a6-9c12-4c64-b241-5f860f1ff3c2.png">

`stable 2.5.1 without RendererBinding.instance!.setSematicsEnabled(true)`

<img width="801" alt="Screenshot 2021-09-27 at 13 47 42" src="https://user-images.githubusercontent.com/88313112/134921345-7b49149e-1f08-435c-88b6-fb987c1e09f9.png">

</details>

<details>
<summary>logs</summary>

`stable 2.2.3`

```console
Launching lib/issue_90794.dart on Chrome in debug mode...
This app is linked to the debug service: ws://127.0.0.1:56355/pEN-rS5iWEc%3D/ws
Debug service listening on ws://127.0.0.1:56355/pEN-rS5iWEc=/ws
💪 Running with sound null safety 💪
Connecting to VM Service at ws://127.0.0.1:56355/pEN-rS5iWEc=/ws
Entered: h
Entered: he
Entered: hel
Entered: hell
Entered: hello
Entered: hello
Entered: hello f
Entered: hello fl
Entered: hello flu
Entered: hello flut
Entered: hello flutt
Entered: hello flutte
Entered: hello flutter
Entered: hello flutte
Entered: hello flutt
Entered: hello flut
Entered: hello flu
Entered: hello fl
Entered: hello f
Entered: hello
Entered: hello
Entered: hell
Entered: hel
Entered: he
Entered: h
Entered:
Entered: h
Entered: he
Entered: hel
Entered: hell
Entered: hello
Application finished.
Exited (sigterm)

```

`stable 2.5.1 & master`

```console
Launching lib/issue_90794.dart on Chrome in debug mode...
This app is linked to the debug service: ws://127.0.0.1:56051/o9ONn9pW2Us%3D/ws
Debug service listening on ws://127.0.0.1:56051/o9ONn9pW2Us=/ws
💪 Running with sound null safety 💪
Connecting to VM Service at ws://127.0.0.1:56051/o9ONn9pW2Us=/ws
Entered: h
Entered: •e
Entered: ••l
Entered: •••l
Entered: ••••o
Entered: ••••
Entered: •••
Entered: ••
Entered: •
Entered:
Entered: h
Entered: •e
Entered: ••l
Entered: •••l
Entered: ••••o
Entered: •••••
Entered: ••••••f
Entered: •••••••l
Entered: ••••••••u
Entered: •••••••••t
Entered: ••••••••••t
Entered: •••••••••••e
Entered: ••••••••••••r
Application finished.
Exited (sigterm)

```

`stable 2.5.1 without RendererBinding.instance!.setSematicsEnabled(true)`

```console
Launching lib/issue_90794.dart on Chrome in debug mode...
This app is linked to the debug service: ws://127.0.0.1:57375/oc6NQweYc6g%3D/ws
Debug service listening on ws://127.0.0.1:57375/oc6NQweYc6g=/ws
💪 Running with sound null safety 💪
Connecting to VM Service at ws://127.0.0.1:57375/oc6NQweYc6g=/ws
Entered: h
Entered: he
Entered: hel
Entered: hell
Entered: hello
Entered: hello
Entered: hello h
Entered: hello he
Entered: hello h
Entered: hello
Entered: hello
Entered: hello  w
Entered: hello  wo
Entered: hello  wor
Entered: hello  worl
Entered: hello  world
Entered: hello  worl
Entered: hello  wor
Entered: hello  wo
Entered: hello  w
Entered: hello
Entered: hello
Entered: hello
Entered: hell
Entered: hel
Entered: he
Entered: h
Entered:
Application finished.
Exited (sigterm)

```

</details>

<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.2.3 at /Users/nexus/dev/sdks/flutter_rc/flutter
    • Framework revision f4abaa0735 (3 months ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS 14.8
    • macOS (desktop)        • macos                     • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome 93.0.4577.82

• No issues found!
```

```console
[✓] Flutter (Channel stable, 2.5.1, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.5.1 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (10 days ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS 14.8 18H17
    • macOS (desktop)        • macos                     • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome 93.0.4577.82

• No issues found!
```

```console
[✓] Flutter (Channel master, 2.6.0-12.0.pre.98, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-12.0.pre.98 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 48e6758fb5 (12 hours ago), 2021-09-26 21:28:03 -0400
    • Engine revision 65ad5a4836
    • Dart version 2.15.0 (build 2.15.0-150.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS 14.8 18H17
    • macOS (desktop)        • macos                     • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome 93.0.4577.82

• No issues found!
```

</details>

**Comment by mdebbar** [comment]
I did a little investigation. This is happening here:

https://github.com/flutter/engine/blob/eb97bda626f7c7bf93248bd460947193978825c2/lib/web_ui/lib/src/engine/semantics/text_field.dart#L351-L355

The `value` in the semantic object is obscured (i.e. "•••" instead of "abc"). I think the obscured string is coming from here:

https://github.com/flutter/flutter/blob/afd84ad425b870cd77b26f1838ad3e2e2f0fdb1f/packages/flutter/lib/src/rendering/editable.dart#L1151-L1153

@yjbanov any thoughts?

**Comment by yjbanov** [comment]
Is the semantic value obscured to prevent the assistive technology from reading the value out loud? If so, perhaps we should store the actual value in a separate field that we know won't be read by the AT.

**Comment by ditman** [comment]
Maybe related? https://stackoverflow.com/questions/48713912/screen-readers-reads-full-stop-or-star-for-password-type-input-on-mobile-app/48736897#48736897

**Comment by chrisnorman7** [comment]
Hi,
This bug still seems to be present, but only after enabling semantics programatically:

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    final instance = RendererBinding.instance;
    if (instance != null) {
      instance.setSemanticsEnabled(true);
    } else {
      throw Exception('No instance.');
    }
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  /// Create an instance.
  const MyHomePage({Key? key}) : super(key: key);

  /// Create state for this widget.
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

/// State for [MyHomePage].
class _MyHomePageState extends State<MyHomePage> {
  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
  final TextEditingController _controller = TextEditingController();

  /// Build a widget.
  @override
  Widget build(BuildContext context) {
    final text = _controller.text;
    final Widget child;
    if (text.isEmpty) {
      child = Form(
        child: Column(
          children: [
            TextFormField(
              autofocus: true,
              controller: _controller,
              decoration: const InputDecoration(labelText: 'Password'),
              obscureText: true,
            )
          ],
        ),
        key: _formKey,
      );
    } else {
      child = Center(
        child: Text(text),
      );
    }
    return Scaffold(
      appBar: AppBar(
        title: const Text('Password Fields'),
      ),
      body: child,
      floatingActionButton: FloatingActionButton(
        onPressed: () => setState(() {
          if (text.isNotEmpty) {
            _controller.text = '';
          }
        }),
        tooltip: text.isEmpty ? 'Submit' : 'Clear',
      ),
    );
  }
}
```

If you comment out the line `instance.setSemanticsEnabled(true);`, then the passwords work as expected.

Output of `flutter doctor -v`:

```shell
[ÔêÜ] Flutter (Channel master, 2.9.0-1.0.pre.299, on Microsoft Windows [Version 10.0.22000.376], locale en-GB)
    ÔÇó Flutter version 2.9.0-1.0.pre.299 at C:\Users\chris\flutter
    ÔÇó Upstream repository https://github.com/flutter/flutter.git
    ÔÇó Framework revision 098721e341 (6 hours ago), 2022-01-10 05:40:18 -0500
    ÔÇó Engine revision 6a9c04dc47
    ÔÇó Dart version 2.16.0 (build 2.16.0-144.0.dev)
    ÔÇó DevTools version 2.9.2
    ÔÇó PATH = C:\tools\ruby31\bin;C:\Program Files\Microsoft\jdk-11.0.12.7-hotspot\bin;C:\tools\ruby30\bin;C:\Python310\Scripts\;C:\Python310\;C:\Python39\Scripts\;C:\Python39\;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\Program Files\Microsoft VS Code\bin;C:\ProgramData\chocolatey\bin;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\LLVM\bin;C:\HashiCorp\Vagrant\bin;C:\Program Files (x86)\Lua\5.1;C:\Program Files (x86)\Lua\5.1\clibs;C:\Program Files\Git\cmd;C:\Program Files\nodejs\;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files (x86)\IncrediBuild;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\dotnet\;C:\Program Files\Go\bin;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build;C:\Users\chris\AppData\Local\Microsoft\WindowsApps;C:\Program Files (x86)\GitHub CLI\;C:\Program Files (x86)\Nmap;C:\Program Files\Git\usr\bin;C:\Users\chris\flutter\bin;C:\Users\chris\flutter\bin\cache\dart-sdk\bin;C:\Users\chris\AppData\Local\Pub\Cache\bin;C:\Users\chris\.dotnet\tools;C:\Program Files\Oracle\VirtualBox;C:\Program Files\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build;C:\tools\cygwin\bin;C:\Users\chris\AppData\Local\Pandoc\;C:\Program Files\Android\Android Studio\jre\bin;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\Extensions\Xamarin.VisualStudio\7-Zip;C:\Program Files (x86)\NSIS;C:\Users\chris\AppData\Roaming\npm;C:\Program Files\eSpeak NG;c:\festival\src\main;C:\Users\chris\go\bin

[ÔêÜ] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc2)
    ÔÇó Android SDK at C:\Users\chris\AppData\Local\Android\sdk
    ÔÇó Platform android-31, build-tools 31.0.0-rc2
    ÔÇó Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    ÔÇó Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    ÔÇó All Android licenses accepted.

[ÔêÜ] Chrome - develop for the web
    ÔÇó Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[ÔêÜ] Visual Studio - develop for Windows (Visual Studio Community 2022 17.0.4)
    ÔÇó Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    ÔÇó Visual Studio Community 2022 version 17.0.32014.148
    ÔÇó Windows 10 SDK version 10.0.22000.0

[ÔêÜ] Android Studio (version 2020.3)
    ÔÇó Android Studio at C:\Program Files\Android\Android Studio
    ÔÇó Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    ÔÇó Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    ÔÇó Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[ÔêÜ] VS Code, 64-bit edition (version 1.63.2)
    ÔÇó VS Code at C:\Program Files\Microsoft VS Code
    ÔÇó Flutter extension version 3.27.0

[ÔêÜ] Connected device (3 available)
    ÔÇó Windows (desktop) ÔÇó windows ÔÇó windows-x64    ÔÇó Microsoft Windows [Version 10.0.22000.376]
    ÔÇó Chrome (web)      ÔÇó chrome  ÔÇó web-javascript ÔÇó Google Chrome 96.0.4664.110
    ÔÇó Edge (web)        ÔÇó edge    ÔÇó web-javascript ÔÇó Microsoft Edge 96.0.1054.62

[ÔêÜ] HTTP Host Availability
    ÔÇó All required HTTP hosts are available

ÔÇó No issues found!
```

**Comment by chrisnorman7** [comment]
So apparently I'm wrong. It's still definitely there, unless you use the above code to show a text field on the top-level widget.

The below code (run on the same version of Flutter) which requires you to click a button before the password field shows still gives you bullet marks.

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    final instance = RendererBinding.instance;
    if (instance != null) {
      // instance.setSemanticsEnabled(true);
    } else {
      throw Exception('No instance.');
    }
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  /// Create an instance.
  const MyHomePage({Key? key}) : super(key: key);

  /// Create state for this widget.
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

/// State for [MyHomePage].
class _MyHomePageState extends State<MyHomePage> {
  /// Build a widget.
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Landing Page'),
      ),
      body: const Center(
        child: Text('Click the floating button.'),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () => Navigator.of(context).push(
          MaterialPageRoute<void>(
            builder: (context) => const MyPasswordPage(),
          ),
        ),
      ),
    );
  }
}

class MyPasswordPage extends StatefulWidget {
  /// Create an instance.
  const MyPasswordPage({Key? key}) : super(key: key);

  /// Create state for this widget.
  @override
  _MyPasswordPageState createState() => _MyPasswordPageState();
}

/// State for [MyPasswordPage].
class _MyPasswordPageState extends State<MyPasswordPage> {
  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
  final TextEditingController _controller = TextEditingController();

  /// Build a widget.
  @override
  Widget build(BuildContext context) {
    final text = _controller.text;
    final Widget child;
    if (text.isEmpty) {
      child = Form(
        child: Column(
          children: [
            TextFormField(
              autofocus: true,
              controller: _controller,
              decoration: const InputDecoration(labelText: 'Password'),
              obscureText: true,
            )
          ],
        ),
        key: _formKey,
      );
    } else {
      child = Center(
        child: Text(text),
      );
    }
    return Scaffold(
      appBar: AppBar(
        title: const Text('Password Fields'),
      ),
      body: child,
      floatingActionButton: FloatingActionButton(
        onPressed: () => setState(() {
          if (text.isNotEmpty) {
            _controller.text = '';
          }
        }),
        tooltip: text.isEmpty ? 'Submit' : 'Clear',
      ),
    );
  }
}
```

**Comment by ih-Semjon-Liakhov** [comment]
Still reproducible on Flutter 2.10.2

**Comment by GiacomoPignoni** [comment]
Still present on Flutter 2.10.4
Is there some update about it?

**Comment by alexlomm-dormzi** [comment]
`RendererBinding.instance!.setSemanticsEnabled(true);` also seems to be making keyboard not show app when `TextFormField` is focused when the app is accessed from browsers in IOS.

**Comment by thailocnexle** [comment]
still in 2.10.5

**Comment by alexlindroos** [comment]
Still in 3.0.5

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-x64, locale en-FI)
    • Flutter version 3.0.5 at /Users/codemate/desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2
```

**Comment by krokyze** [comment]
Still in 3.3.2

**Comment by htoor3** [comment]
> I tested in 3.7.1 and it's still broken. Looks like the fix was reverted: [flutter/engine#38562](https://github.com/flutter/engine/pull/38562) Where can I track this being fixed going forward?

@nhinmanBhg the changes were remerged here https://github.com/flutter/engine/pull/38854 - I retested this issue with the above code sample on `master` and confirmed the issue is fixed.  I believe the change should land in the next stable release
